lib/src/date_extension.dart

```diff
@@ -0,0 +1,65 @@
+/// The code snippet defines a private class `_DateExtension` that takes two `DateTime` parameters:
+/// `currentTime` and `dateTime`. It also defines two private variables `_currentTime` and `_dateTime`
+/// to store these parameters.
+final class _DateExtension {
+  _DateExtension({
+    required DateTime currentTime,
+    required DateTime dateTime,
+    required DateDifferenceModel dateDifferenceModel,
+  })  : _currentTime = currentTime,
+        _dateTime = dateTime,
+        _dateDifferenceModel = dateDifferenceModel;
+  final DateTime _currentTime;
+  final DateTime _dateTime;
+  final DateDifferenceModel _dateDifferenceModel;
+
+  /// The `differenceTime` getter is a method that calculates the time difference between the current
+  /// DateTime object and the current time.
+  String get differenceTime {
+    final dayDifference = _currentTime.difference(_dateTime).inDays;
+    final hourDifference = _currentTime.difference(_dateTime).inHours;
+    final minuteDifference = _currentTime.difference(_dateTime).inMinutes;
+    if (dayDifference > 0) {
+      return '$dayDifference ${_dateDifferenceModel.dayDifferenceLabel}}';
+    } else if (hourDifference > 0) {
+      return '$hourDifference ${_dateDifferenceModel.hourDifferenceLabel}';
+    } else if (minuteDifference > 0) {
+      return '$minuteDifference ${_dateDifferenceModel.minuteDifferenceLabel}';
+    }
+    return '';
+  }
```

Comment:
change to business beucase it not accaptable. For example dayDifrrence is bigger than zero, your code trying to parse every diff for each item. This need to calculate after the check business

final dayDifference = _currentTime.difference(_dateTime).inDays;
example
if(day > 0) return '';